owncloudApp/src/androidTest/java/com/owncloud/android/settings/security/OCSettingsPatternLock.kt

```diff
@@ -0,0 +1,100 @@
+/*
+ * ownCloud Android client application
+ *
+ * @author Jesus Recio (@jesmrec)
+ * Copyright (C) 2019 ownCloud GmbH.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.owncloud.android.settings.security
+
+import android.content.Intent
+import android.preference.PreferenceManager
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.rule.ActivityTestRule
+import com.owncloud.android.R
+import com.owncloud.android.ui.activity.PassCodeActivity
+import com.owncloud.android.ui.activity.PatternLockActivity
+import org.junit.After
+import org.junit.Assert.assertTrue
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class OCSettingsPatternLock {
+
+    @Rule
+    @JvmField
+    val activityRule = ActivityTestRule(PatternLockActivity::class.java, true, false)
+    val errorMessage = "PatternLock Activity error"
+    val intent = Intent()
+    private val context = InstrumentationRegistry.getInstrumentation().targetContext
+
+    @After
+    fun tearDown() {
+        //Clean preferences
+        PreferenceManager.getDefaultSharedPreferences(context).edit().clear().commit()
+    }
+
+    @Test
+    fun patternLockView(){
+        //Open Activity in pattern creation mode
+        intent.setAction(PatternLockActivity.ACTION_REQUEST_WITH_RESULT)
```

Comment:
Use this method instead?

```
private fun openPasscodeActivity (mode: string) {
    intent.setAction(mode)
    activityRule.launchActivity(intent)
} 
```

Reply:
Done 👍

Reply:
Ops, I didn't write `intent.action = mode` in my example

Reply:
it is already done

Reply:
It's done in `OCSettingsPasscodeTest` class but the method `openPatternActivity()` of `OCSettingsPatternLockTest` still has it.

Reply:
Done 👍